base/sim/FairModule.h

```diff
@@ -168,51 +179,59 @@ class FairModule : public TNamed
     ClassDefOverride(FairModule, 4);
 };
 
-template<class T, class U>
-void FairModule::ConstructASCIIGeometry(T* dataType1, TString containerName, U*)
+template<typename GeoModule>
+GeoModule* FairModule::ConstructASCIIGeometryHelper()
 {
-    FairGeoLoader* loader = FairGeoLoader::Instance();
-    FairGeoInterface* GeoInterface = loader->getGeoInterface();
-    T* MGeo = new T();
-    MGeo->print();
-    MGeo->setGeomFile(GetGeometryFileName());
-    GeoInterface->addGeoModule(MGeo);
-    Bool_t rc = GeoInterface->readSet(MGeo);
-    if (rc) {
-        MGeo->create(loader->getGeoBuilder());
+    auto loader = FairGeoLoader::Instance();
+    auto geoInterface = loader->getGeoInterface();
+    auto geoModule = static_cast<GeoModule*>(geoInterface->addGeoModule(std::make_unique<GeoModule>()));
+    geoModule->setGeomFile(GetGeometryFileName());
+    geoModule->print();
+    if (geoInterface->readSet(geoModule)) {
+        geoModule->create(loader->getGeoBuilder());
     }
+    return geoModule;
+}
 
-    TList* volList = MGeo->getListOfVolumes();
-    // store geo parameter
-    FairRun* fRun = FairRun::Instance();
-    FairRuntimeDb* rtdb = FairRun::Instance()->GetRuntimeDb();
+template<typename GeoModule, typename ParameterSet>
+void FairModule::AddGeoNodesToParameterSetHelper(GeoModule& geoModule, std::string const& parameterSetName)
```

Comment:
```suggestion
template<typename ParameterSet, typename GeoModule>
void FairModule::AddGeoNodesToParameterSetHelper(std::string const& parameterSetName, GeoModule& geoModule)
```

Reply:
Ideally, it should also be `GeoModule const& geoModule` parameter, but this requires `getListOfVolumes()` to be `const` as well